index.bs

```diff
@@ -1168,6 +1168,33 @@ The authenticator MUST return both the <a>authenticatorData</a> and the raw sign
 MUST return <a>clientDataJSON</a>, <a>authenticatorData</a> and the signature to the RP. The first two are returned in the
 `clientData` and `authenticatorData` members respectively of the {{WebAuthnAssertion}} and {{WebAuthnAttestation}} structures.
 
+### Verifying a signature ### {#authenticator-signature-verification}
+
+This section specifies the algorithm for verifying a signature assertion.
+
+Upon receiving a signature assertion in the form of a {{WebAuthnAssertion}} structure, the [RP] shall:
+
+1. Perform JSON decoding to extract the {{ClientData}} used for the assertion from the {{WebAuthnAssertion/clientData}}.
+
+2. Verify that the {{ClientData/challenge}} in the {{ClientData}} matches the challenge that was sent to the authenticator.
+
+3. Verify that the {{ClientData/origin}} in the {{ClientData}} matches the [RP]'s origin.
+
+4. Verify that the {{ClientData/tokenBinding}} in the {{ClientData}} matches the token binding public key for the TLS connection
+    over which the attestation was obtained.
+
+5. Verify that the {{ClientData/extensions}} in the {{ClientData}} is a proper subset of the extensions requested by the RP.
+
+6. Verify that the RP ID hash in the {{WebAuthnAssertion/authenticatorData}} is indeed the SHA-256 hash of the RP ID expected
+    by the RP.
+
+7. Compute the <a>clientDataHash</a>, i.e. hash of {{WebAuthnAssertion/clientData}}.
+
+8. Look up the public key that was returned by {{makeCredential()}} and is related to {{WebAuthnAssertion/credential}} and
+    verify the signature in {{WebAuthnAssertion/signature}} computed over the
+    binary concatenation of {{WebAuthnAssertion/authenticatorData}} and <a>clientDataHash</a>.
+
+The Relying Party shall reject the request if the verification fails.
```

Comment:
I would stick to the verification aspects here. The RP will do what it will do.

So I'd suggest replacing this with something like "if all the above steps succeed, then the signature is valid, otherwise not".